modules/platform/nuxeo-automation/nuxeo-automation-test/src/test/java/org/nuxeo/ecm/automation/server/test/AsyncOperationAdapterTest.java

```diff
@@ -208,6 +208,18 @@ public void testError() throws Exception {
                             .set("rollback", true)
                             .executeReturningExceptionEntity(ExitOperation.ERR_CODE);
         assertEquals("Failed to invoke operation Test.Exit, termination error", error);
+
+        // automation bulk action with invalid query
+        Map<String, Serializable> automationParams = Map.of("properties", "dc:title=folder_title_updated");
+        Map<String, Serializable> actionParams = Map.of(AutomationBulkAction.OPERATION_ID, "Document.Update", //
+                AutomationBulkAction.OPERATION_PARAMETERS, (Serializable) automationParams);
+        JsonNode jsonNode = async.newRequest(BulkRunAction.ID)
+                                 .set("action", AutomationBulkAction.ACTION_NAME)
+                                 .set("query", "SELCT * FROM Folder") // Invalid Query
+                                 .set("bucketSize", "10")
+                                 .set("batchSize", "5")
+                                 .set("parameters", MAPPER.writeValueAsString(actionParams))
+                                 .execute(SC_INTERNAL_SERVER_ERROR);
```

Comment:
I don't like that this is returning `SC_INTERNAL_SERVER_ERROR`; the latest changes were careful to return `SC_BAD_REQUEST` for all errors that are due to user input.

But this may mean slightly deeper changes to have `BulkStatus.inError()` take and record also an error status, and change its callers to provide one when available. So that typically `BulkScrollerComputation.processRecord` can provide one when calling `updateStatusAfterScroll` for an invalid query. And `AbstractBulkComputation.processFailure` should check if the `failure` is a `NuxeoException` and reuse its status code in that case.

Reply:
I agree On that. And, as you has mentioned it, It needs a deeper changes on Bulk part. I'll go deeper and rework things.